wiki_articles/cas.md

```diff
@@ -0,0 +1,37 @@
+# What Is an Atomic Compare-And-Swap (CAS)?
+
+[CAS](https://en.wikipedia.org/wiki/Compare-and-swap)
+lets us implement
+*[read-modify-write operations](https://en.wikipedia.org/wiki/Read%E2%80%93modify%E2%80%93write)*
+for atomics,
+no matter how complex modifying is.
+It's often the foundation of
+*[lock-free algorithms](https://en.wikipedia.org/wiki/Non-blocking_algorithm)*.
+In C++, this takes the form of
+**[std::atomic::compare_exchange_xxx](https://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange)**:
```

Comment:
I think this is a little editorialized

Reply:
How is this expressing opinion? What opinion? Why did you highlight everything from line 3 to 11?

Do you think it's an opinion that CAS takes the form of `std::atomic_compare_exchange` in C++?